### PR TITLE
modify engine.py for formatting

### LINKED ISSUE #2666: [BUG] some docs have broken formatting
**Describe the bug**

The API arguments docs aren't formatted, e.g. these ones (but there are probably more of those):

https://deepspeed.readthedocs.io/en/latest/training.html#model-saving
https://deepspeed.readthedocs.io/en/latest/training.html#gradient-accumulation
https://deepspeed.readthedocs.io/en/latest/model-checkpointing.html (most of the page)

e.g. have a look at: https://deepspeed.readthedocs.io/en/latest/model-checkpointing.html#loading-training-checkpoints

here all the args are piled into one para, instead of being itemized and nothing is formatted:

> Load training checkpoint :param load_dir: Required. Directory to load the checkpoint from :param tag: Checkpoint tag used as a unique identifier for checkpoint, if not provided will attempt to load tag in ‘latest’ file :param load_module_strict: Optional. Boolean to strictly enforce that the keys in state_dict of module and checkpoint match. :param load_optimizer_states: Optional. Boolean to load the training optimizer states from Checkpoint. Ex. ADAM’s momentum and variance :param load_lr_scheduler_states: Optional. Boolean to add the learning rate scheduler states from Checkpoint. :param load_module_only: Optional. Boolean to load only the model weights from the checkpoint. Ex. warmstarting. :param custom_load_fn: Optional. Custom model load function.

### PR DESCRIPTION
**Pull request by xiexbing**
fix #2666

### CODE REVIEW

**Review by tjruwase — APPROVED** [review]

**Comment by xiexbing** [comment]
sorry, I think I wrongly committed changes to another branch and create a new pull request #2764. I'll close this PR and please approve the new one intead. thanks
